client/components/AddSponsor.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import styled from 'styled-components';
+
+interface AddSponsorProps {
+  eventId: string;
+  chapterId: string;
+}
+
+interface SponsorData {
+  name: string;
+  website: string;
+  type: 'FOOD' | 'BEVERAGE' | 'OTHER'; // TODO: Add VENUE
+}
+
+const Form = styled.form`
+  display: flex;
+  flex-direction: column;
+  max-width: 25%;
+`;
+
+const Input = styled.input`
+  margin-bottom: 15px;
+`;
+
+const ResponseDiv = styled.div`
+  margin: 15px 0;
+`;
+// TODO: Style div based on response reveived
+
+const AddSponsor: React.FC<AddSponsorProps> = ({ eventId, chapterId }) => {
+  const [isSubmitting, setSubmitting] = React.useState(false);
+  // we specify type as food as state is only updated on field change. Since field
+  // defaults to food and the user doesn't change it, state would be empty
+  const [values, setValues] = React.useState<SponsorData>({
+    name: '',
+    website: '',
+    type: 'FOOD',
+  });
+  const [success, setSuccess] = React.useState(false);
+
+  function handleChange(e) {
+    const { value, name } = e.target;
+    setValues({
+      ...values,
+      [name]: value,
+    });
+  }
+
+  async function handleSubmit(e) {
+    /* Block double submits */
+    if (isSubmitting) {
+      return false;
+    }
+
+    e.preventDefault();
+    setSubmitting(true);
+
+    try {
+      const data = JSON.stringify(values);
+      await fetch(`/${chapterId}/events/${eventId}/sponsors`, {
```

Comment:
seeing `'fetch' is not defined` - can you install the npm package `isomorphic-fetch` so you can `import fetch from 'isomorphic-fetch`?  we're gonna need this anyway when doing more server-rendered stuff

Reply:
I know that we need `isomorphic-fetch` for `getInitalProps`, because that is done on the server side, but this will always be done on the client, so do we need `isomorphic-fetch` here or should the normal `fetch` be fine for this?

Reply:
Not needed, although we _could_ use isomorphic fetch here for consistency.  Shouldn't matter either way